packages/block-library/src/file/edit.js

```diff
@@ -116,15 +112,14 @@ function FileEdit( {
 	}, [] );
 
 	useEffect( () => {
-		if ( ! fileId ) {
+		if ( ! fileId && href ) {
```

Comment:
This fixes the "undo trap" issue in the placeholder state. The change was small, so I decided to include it in this PR, but I can create a separate one to make reviewing easier.

Reply:
Should we also dispatch `__unstableMarkNextChangeAsNotPersistent` in the effect?

Reply:
I don't think this is needed based on my tests.

Reply:
I just tested with and without it and it seemed to make a difference when you add upload valid file:
1. Add a file block
2. Drag a valid file type
3. Try pressing undo a few times

without `__unstableMarkNextChangeAsNotPersistent` - undo gets 'stuck.
with `__unstableMarkNextChangeAsNotPersistent` - undo works correctly.

Reply:
Hmm, I guess I missed something. I will update the code and add action. Thanks for testing.

Reply:
I'm happy to push that change as a separate PR or as a commit to this branch. You have approval and this is really an unrelated thing to the bug you were fixing.

Let me know what you think is best.

Reply:
I think I'll create separate PR for "undo trap" issues (I noticed another one); it would be better for git history.

I will revert this part of the code from this PR as well.

Reply:
Sounds good, feel free to add me as a reviewer. 👍